packages/fuels-programs/src/calls/call_handler.rs

```diff
@@ -134,41 +119,11 @@ where
 impl<A, C, T> CallHandler<A, C, T>
 where
     A: Account,
-    C: ContractDependencyConfigurator + TransactionTuner + ResponseParser,
+    C: TransactionTuner + ResponseParser,
     T: Tokenizable + Parameterize + Debug,
 {
-    /// Sets external contracts as dependencies to this contract's call.
-    /// Effectively, this will be used to create [`fuel_tx::Input::Contract`]/[`fuel_tx::Output::Contract`]
-    /// pairs and set them into the transaction. Note that this is a builder
-    /// method, i.e. use it as a chain:
-    ///
-    /// ```ignore
-    /// my_contract_instance.my_method(...).with_contract_ids(&[another_contract_id]).call()
-    /// ```
-    ///
-    /// [`Input::Contract`]: fuel_tx::Input::Contract
-    /// [`Output::Contract`]: fuel_tx::Output::Contract
-    pub fn with_contract_ids(mut self, contract_ids: &[Bech32ContractId]) -> Self {
-        self.call = self.call.with_external_contracts(contract_ids.to_vec());
-
-        self
-    }
-
-    /// Sets external contract instances as dependencies to this contract's call.
-    /// Effectively, this will be used to: merge `LogDecoder`s and create
-    /// [`fuel_tx::Input::Contract`]/[`fuel_tx::Output::Contract`] pairs and set them into the transaction.
-    /// Note that this is a builder method, i.e. use it as a chain:
-    ///
-    /// ```ignore
-    /// my_contract_instance.my_method(...).with_contracts(&[another_contract_instance]).call()
-    /// ```
-    pub fn with_contracts(mut self, contracts: &[&dyn ContractDependency]) -> Self {
-        self.call = self
-            .call
-            .with_external_contracts(contracts.iter().map(|c| c.id()).collect());
-        for c in contracts {
-            self.log_decoder.merge(c.log_decoder());
-        }
+    pub fn add_log_decoder(mut self, log_decoder: LogDecoder) -> Self {
```

Comment:
Can we document on that. What is a `LogDecoder` could be interesting also because I don't see what is it.

Reply:
e2096da0